### DISCUSSION TITLE
Create virtual metaData from create table.

### DISCUSSION THREAD

**Discussion by jtknights08** [sqlalchemy/alembic#1408] [Usage Questions] [answered]
Is there currently a way to give alembic a create table statement or parsing mechanism as the metaData to update a target db? This is a weekly backup we want to automate the schema updates for and currently use Alembic for the source data these backups come from but we can't guarantee the schema is accurate to our files.

**Comment by zzzeek** [comment]
this is the basic use case for autogenerate:

https://alembic.sqlalchemy.org/en/latest/autogenerate.html

**Reply by jtknights08** [reply]
@zzzeek Thanks for the quick response, I checked over that file and did not see a way to generate a metaData from a raw create statement? Here is an example of what I'm talking about:

```
CREATE TABLE 'Pizza' (
  'size' varchar(2) NOT NULL,
  'toppings' varchar(20) NOT NULL,
  'customer' varchar(2) NOT NULL,
  'price' varchar(20) NOT NULL,
  'idx' bigint(20) unsigned NOT NULL AUTO_INCREMENT,
  PRIMARY KEY ('idx'),
  UNIQUE KEY 'idType' ('customer','idx'),
  KEY 'customer' ('customer')
) ENGINE=InnoDB AUTO_INCREMENT=0 DEFAULT CHARSET=latin1;
```

**Reply by zzzeek** [reply, marked as answer]
you would have to run those CREATE statements into a database somewhere, then reflect them back.

if you dont want a database involved then you'd have to do some kind of parsing.  The MySQL dialect in SQLAlchemy does have a parser for CREATE TABLE internally so perhaps you can re-purpose it.
